storage_mem.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"fmt"
+
+	model "github.com/golang-bristol/beer-model"
+)
+
+// StorageMemory data storage type save only in memory
+type StorageMemory struct {
+	cellar  []model.Beer
+	reviews []model.Review
+}
+
+func newStorageMemory() (*StorageMemory, error) {
+	var err error
+
+	stg := new(StorageMemory)
+
+	return stg, err
+}
+
+// SaveBeer insert or update beers
+func (s *StorageMemory) SaveBeer(beers ...model.Beer) error {
+	for _, beer := range beers {
+		var err error
+
+		beersFound, err := s.FindBeer(model.Beer{ID: beer.ID})
+		if err != nil {
+			return err
+		}
+
+		if len(beersFound) == 1 {
+			*beersFound[0] = beer
```

Comment:
Not sure why we need to do this here?

Reply:
This sets a new value for an existing beer in the cellar field array. 

I thought it would be better to reuse findBeer() results to find and change current values of the struct fields.

Reply:
Hmm, I don't think this will work the way you think it will. FindBeer returns you a slice of pointers to beer models, but it's not tied in any way to the cellar variable. So changing the beersFound slice here will not overwrite the value in the original cellar variable. Unless that's not what you meant here?

Reply:
My bad. I messed up. I'll fix this.